service/storage_service.cc

```diff
@@ -714,14 +714,14 @@ class topology_coordinator {
     future<> cleanup_group0_config_if_needed() {
         auto& topo = _topo_sm._topology;
         auto rconf = _group0.group0_server().get_configuration();
-        if (rconf.current.size() > topo.normal_nodes.size() + topo.new_nodes.size() + topo.transition_nodes.size()) {
-            // Raft config is larger than the sum of all nodes not in 'left' state.
+        auto to_remove = boost::copy_range<std::unordered_set<raft::server_id>>(
+                boost::range::join(rconf.current, rconf.previous)
+                | boost::adaptors::transformed([&] (const raft::config_member& m) { return m.addr.id; })
+                | boost::adaptors::filtered([&] (const raft::server_id& id) { return topo.left_nodes.contains(id); }));
+        if (!to_remove.empty()) {
             // Find nodes that 'left' but still in the config and remove them
-            size_t found = 0;
-            co_await coroutine::parallel_for_each(topo.left_nodes | boost::adaptors::filtered(
-                    [&rconf] (raft::server_id id) { return rconf.contains(id); }),
+            co_await coroutine::parallel_for_each(to_remove,
```

Comment:
The previous code used `parallel_for_each`, but it is not really needed since we can do everything in one call to `modify_config`. Just pass entire `to_remove` to it. I planned to change it, but forgot.

Reply:
Good idea, done